kolibri/plugins/management/assets/src/views/user-page/user-create-modal.vue

```diff
@@ -119,61 +117,70 @@
+      },
+      usernameNotAlphaNum() {
+        if (this.username === '') {
+          return false;
+        }
+        const re = /^\w+$/g;
+        return !re.test(this.username);
+      },
+      usernameInvalid() {
+        if (this.usernameAlreadyExists) {
+          return true;
+        } else if (this.usernameNotAlphaNum) {
+          return true;
         }
         return false;
       },
+      usernameInavlidMsg() {
+        if (this.usernameAlreadyExists) {
+          return this.$tr('usernameAlreadyExists');
+        } else if (this.usernameNotAlphaNum) {
+          return this.$tr('usernameNotAlphaNum');
+        }
+        return '';
+      },
+      passwordConfirmInvalid() {
+        if (this.passwordConfirm === '') {
+          return false;
+        }
+        return this.password !== this.passwordConfirm;
+      },
     },
     methods: {
       createNewUser() {
-        const newUser = {
-          username: this.username,
-          full_name: this.fullName,
-          kind: this.kind,
-        };
-
-        // check for all fields populated
-        if (!(this.username && this.password && this.fullName && this.kind)) {
-          this.errorMessage = this.$tr('emptyFieldError');
-        // check for password confirmation match
-        } else if (!(this.password === this.passwordConfirm)) {
-          this.errorMessage = this.$tr('pwMismatchError');
-        // create user
-        } else {
-          newUser.password = this.password;
-
-          // loading message
-          this.confirmationMessage = this.$tr('loadingConfirmation');
+        this.errorMessage = '';
+        if (!this.usernameInvalid && !this.passwordConfirmInvalid) {
+          this.loading = true;
+
+          const newUser = {
+            username: this.username,
+            full_name: this.fullName,
+            kind: this.kind,
+            password: this.password,
+          };
           // using promise to ensure that the user is created before closing
           this.createUser(newUser).then(
             () => {
               this.close();
             },
-            (error) => {
-              if (error.status.code === 400) {
-                // access the first error message
-                this.errorMessage = error.entity[Object.keys(error.entity)[0]];
-              } else if (error.status.code === 403) {
-                this.errorMessage = error.entity;
-              } else {
-                this.errorMessage = this.$tr('unknownError');
-              }
+            error => {
+              this.loading = false;
+              this.errorMessage = this.$tr('unknownError');
             });
         }
       },
-      clearStatus() {
-        this.errorMessage = '';
-        this.confirmationMessage = '';
-      },
       close() {
         this.displayModal(false);
       },
     },
     vuex: {
+      getters: {
+        users: state => state.pageState.facilityUsers,
```

Comment:
Added the error catching messages back in.
![image](https://cloud.githubusercontent.com/assets/7193975/25598856/6a01ea9c-2e8c-11e7-8c7f-c2fa8da630b4.png)